src/models/image-map.ts

```diff
@@ -171,41 +173,40 @@ export const ImageMapModel = types
         _db = db;
       },
 
-      addFileImage(file: File): Promise<ImageMapEntryType> {
-        return new Promise((resolve, reject) => {
-          storeFileImage(_db, file)
-            .then(simpleImage => {
-              const { normalized } = parseFauxFirebaseRTDBUrl(simpleImage.imageUrl);
-              const entry: ImageMapEntrySnapshot = {
-                      filename: file.name,
-                      contentUrl: normalized,
-                      displayUrl: simpleImage.imageData
-                    };
-              resolve(self.addImage(entry.contentUrl!, entry));
-            });
-        });
+      // General Note: Typically actions should not be async like this. An async action will 
+      // run outside normal action path so MST can't record its changes to the model.
+      // If it tries to modify the model directly MST will raise an error. In the cases
+      // below other actions are always called to mae modifications so this is safe.
```

Comment:
```suggestion
      // below other actions are always called to make modifications so this is safe.
```

Reply:
Do you have an alternative implementation to propose, i.e. should this be a `TODO`?

Reply:
The safe solution is to use a generator with MST `flow`. But it is kind of ugly. And really in this case we don't care so much about the actions. We aren't using any middleware.

Probably the best thing to do here would be to switch to vanilla MobX. But I haven't looked at MobX's support for async. I'll check it out and update the comment.

Reply:
OK I looked into this more. MobX has the same recommend `flow` approach. I had forgotten the other reason to use actions is for transactions/performance. No observers will be notified until the full action is complete. 

Also looking at our code more ImageMap.addImage is actually not safe (in theory). It is setting entry.contentUrl and entry.displayUrl inside of the async code. I think in practice this actually works out because these things happen before any `await` so they are actually run inside of the action call stack. 

I think based on all of this I'll go ahead and switch things to use flow with generators. This will make it safer for future modifications.